Resources/Computing/test/Test_SSHComputingElement.py

```diff
@@ -0,0 +1,63 @@
+#!/bin/env python
+"""
+tests for SSHComputingElement module
+"""
+
+import os
+import time
+import subprocess, shlex
+import pytest
+
+from DIRAC.Resources.Computing.SSHComputingElement import SSHComputingElement
+
+@pytest.mark.parametrize("host", ["Condor", "GE", "Host", "LSF", "OAR", "SLURM", "Torque"])
+def test_generateControlScript(host):
+  """ Test that the control script generated by the merging operation
+      between a BatchSystem and executeBatch.py is executable and doesn't raise any syntax error.
+      
+      Example: it may check that a __future__ import is not misplaced in the script due to the
+      merging of the files. 
+  """
+
+  ce = SSHComputingElement('Test_SSHCE')
+  # Change the batch system file used during the control script generation
+  ce.batchSystem = host
+  # Get the local control script
+  result = ce._generateControlScript()
+  assert result['OK'] is True
+
+  source = result['Value']
+  dest = 'execute_batch.py'
+  
+  # Simulate operation done by the scpCall method
+
+  # Copy the local control script into the "remote" control script
+  # As the source can be composed of multiple files, we have to do it
+  cmd = 'cat %s' % source
+  args = shlex.split(cmd)
+  with open(dest, 'w') as output:
+    process = subprocess.Popen(args, stdout=output)
+    process.communicate()
+  assert process.returncode is 0
+
+  # Add the execution permission on the control script
+  cmd = 'chmod +x %s' % dest
+  args = shlex.split(cmd)
+  process = subprocess.Popen(args)
+  process.communicate()
+  assert process.returncode is 0
+
+  # Test the execution of the remote control script
+  cmd = 'python -m py_compile %s' % dest
+  args = shlex.split(cmd)
+  process = subprocess.Popen(args)
+  process.communicate()
+  assert process.returncode is 0
+
+  # Delete the control script and its .pyc file
+  cmd = 'rm %s %sc' % (dest, dest)
```

Comment:
Apart from the `python` command above, for all the other ones you can use the platform-independent `os` module, or other pure python modules. What you used here is linux only.